lib/realtime/api/message.ex

```diff
@@ -5,16 +5,22 @@ defmodule Realtime.Api.Message do
   use Ecto.Schema
   import Ecto.Changeset
 
+  @primary_key {:id, Ecto.UUID, autogenerate: true}
   @schema_prefix "realtime"
+
   schema "messages" do
     field :topic, :string
     field :extension, Ecto.Enum, values: [:broadcast, :presence]
+    field :payload, :map
+    field :event, :string
+    field :private, :boolean
+
     timestamps()
   end
 
   def changeset(message, attrs) do
     message
-    |> cast(attrs, [:topic, :extension, :inserted_at, :updated_at])
+    |> cast(attrs, [:topic, :extension, :payload, :event, :private, :inserted_at, :updated_at])
```

Comment:
I don't think you need to allow casting timestamps, I think they get overwritten by `put_timestamp`, `maybe_put_timestamp` anyway?

```suggestion
    |> cast(attrs, [:topic, :extension, :payload, :event, :private])
```